src/LinuxTracing/TracerThread.cpp

```diff
@@ -590,12 +617,47 @@ void TracerThread::Startup() {
 
   effective_capture_start_timestamp_ns_ = orbit_base::CaptureTimestampNs();
 
-  // Get the initial thread names and notify the listener_.
+  CaptureStarted capture_started;
+  capture_started.set_process_id(target_pid_);
+
+  auto executable_path_or_error = orbit_base::GetExecutablePath(target_pid_);
+  if (executable_path_or_error.has_value()) {
+    const std::string& executable_path = executable_path_or_error.value();
+    capture_started.set_executable_path(executable_path);
+
+    ErrorMessageOr<std::unique_ptr<orbit_elf_utils::ElfFile>> elf_file_or_error =
+        orbit_elf_utils::ElfFile::Create(executable_path);
+    if (elf_file_or_error.has_value()) {
+      capture_started.set_executable_build_id(elf_file_or_error.value()->GetBuildId());
+    } else {
+      ERROR("Unable to load module: %s", elf_file_or_error.error().message());
+    }
+  } else {
+    ERROR("%s", executable_path_or_error.error().message());
+  }
+
+  auto modules_or_error = orbit_elf_utils::ReadModules(target_pid_);
+  if (modules_or_error.has_value()) {
+    const std::vector<ModuleInfo>& modules = modules_or_error.value();
+    *capture_started.mutable_modules() = {modules.begin(), modules.end()};
+  } else {
+    ERROR("Unable to load modules for %d: %s", target_pid_, modules_or_error.error().message());
+  }
+
+  // Get the initial thread names to construct CaptureStarted event and notify the listener_.
   // All ThreadName events generate by this call will have effective_capture_start_timestamp_ns_ as
   // timestamp. As these events will be the first events of the capture, this prevents later events
   // from having a lower timestamp. After all, the timestamp of the initial ThreadName events is
   // approximate.
-  RetrieveInitialThreadNamesSystemWideAndNotifyListener(effective_capture_start_timestamp_ns_);
+  std::vector<ThreadName> thread_names =
+      RetrieveInitialThreadNamesSystemWide(effective_capture_start_timestamp_ns_);
+
+  *capture_started.mutable_thread_names() = {thread_names.begin(), thread_names.end()};
+
+  capture_started.set_capture_start_timestamp_ns(effective_capture_start_timestamp_ns_);
+  capture_started.mutable_capture_options()->CopyFrom(capture_options_);
+
+  listener_->OnCaptureStarted(std::move(capture_started));
```

Comment:
If this code stays in `TracerThread`, I think that extracting it to `CreateCaptureStartedAndSendToListener` would be useful.

Reply:
moved out of TracerThread